File: 2022/day1.py
with open("input1.txt") as f:
    curr_max = []
    curr_elf = 0
    for i in f.readlines():
        i = i.rstrip()
        if i:
            curr_elf += int(i)
        else:
            curr_max.append(curr_elf)
            curr_elf = 0

sorted_calories = sorted(curr_max, reverse=True)

print(sorted_calories[0])
print(sum(sorted_calories[:3]))

File: 2022/day1.rs
use itertools::Itertools;

fn main() {
    let max: Vec<i32> = include_str!("../input1.txt")
        .lines()
        .map(|line| {
            if line.is_empty() {
                0
            } else {
                line.parse().unwrap()
            }
        })
        .coalesce(|x, y| if y != 0 { Ok(x + y) } else { Err((x, y)) })
        .sorted()
        .rev()
        .collect_vec();

    println!("{:}", max.first().unwrap());
    println!("{:}", max.into_iter().take(3).sum::<i32>());
}
